### PR TITLE
Deprecate `fixed_int_power`

### PR DESCRIPTION
**Pull request by drwells**
I noticed this when I read #5911: we can just use `constexpr` and ~~`std::pow`~~ the new `Utilities::pow` instead of recursive templates, which is nicer. I went ahead and spent a few minutes changing this and marking a few things nearby as `constexpr`.

This passes all of the matrix free tests

### CODE REVIEW

**Comment by masterleinad** [comment]
`ICC-18`  will not like this (https://godbolt.org/g/jfD9vU).

**Comment by masterleinad** [comment]
... and `clang` as well (https://godbolt.org/g/C7TKBh). It seems that `std::pow` being ` constexpr` is up to the compiler and not required.

**Comment by drwells** [comment]
You're right: clang doesn't like this at all. We can get around this by just defining our own `Utilities::pow`.

Lets leave this until #5870 is done since this is a small, mechanical change.

File: include/deal.II/base/utilities.h [hunk 1]
```diff
@@ -302,6 +302,21 @@ namespace Utilities
...
+   * constexpr int value = Utilities::pow(2, dim);
+   * @endcode
+   *
+   * instead of
+   *
+   * @code
+   * const int value = Utilities::fixed_int_power<2, dim>::value;
+   * @endcode
+   *
+   * to obtain a constant expression for <code>value</code>.
```
Comment (Rombur) [inline include/deal.II/base/utilities.h]: Can you ` DEAL_II_DEPRECATED` so that people get a warning when they are using the function

Comment (drwells) [reply include/deal.II/base/utilities.h]: Good point. I fixed it.

**Comment by masterleinad** [comment]
I don't quite understand why `astyle` wants this strange indentation, but it is unhappy.
Maybe put everything on the same line?

**Comment by drwells** [comment]
Yes, that fixes .t

**Pull request merged**